File: CPE-1star/UVA10035-Primary Arithmetic .py
# 計算進位次數
while True:
    try:
        n, m = map(int, input().split())
        if n == 0 and m == 0:
            break
        carry = 0
        while True:
            try:
                if n == 0 and m == 0:
                    break
                if n % 10 + m % 10 >= 10:
                    carry += 1
                    n = n / 10 + 1
                    m /= 10
                else:
                    n /= 10
                    m /= 10
            except:
                break
        if carry == 0:
            print("No carry operation.")
        elif carry == 1:
            print("1 carry operation.")
        else:
            print("%d carry operations." % (carry))
    except:
        break

# V2
while True:
    try:
        a, b = map(int, input().split())
        if a == 0 and b == 0:
            break
        carry = 0
        carrydigit = 0
        while a > 0 or b > 0:
            digitsum = a % 10 + b % 10 + carrydigit
            if digitsum >= 10:
                carry += 1
                carrydigit = 1
            else:
                carrydigit = 0
            a = a // 10
            b = b // 10
        if carry == 0:
            print("No carry operation.")
        elif carry == 1:
            print("1 carry operation.")
        else:
            print("%d carry operations." % (carry))

    except EOFError:
        break

File: CPE-1star/UVA10035-Primary Arithmetic .cpp
#include <iostream>
using namespace std;

int main() {
    while (true) {
        int n, m;
        cin >> n >> m;
        if (n == 0 && m == 0) {
            break;
        }

        int carry = 0;
        int carryDigit = 0;
        while (n > 0 || m > 0) {
            int digitSum = (n % 10) + (m % 10) + carryDigit;
            if (digitSum >= 10) {
                carry++;
                carryDigit = 1;
            } else {
                carryDigit = 0;
            }
            n /= 10;
            m /= 10;
        }

        if (carry == 0) {
            cout << "No carry operation." << endl;
        } else if (carry == 1) {
            cout << "1 carry operation." << endl;
        } else {
            cout << carry << " carry operations." << endl;
        }
    }

    return 0;
}
